test/teams.e2e-spec.ts

```diff
@@ -0,0 +1,189 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { INestApplication, ValidationPipe } from "@nestjs/common";
+import * as request from "supertest";
+import { AppModule } from "../src/app.module";
+import { PrismaService } from "../src/prisma/prisma.service";
+import { seed } from "../prisma/seed/seed";
+import { loginAndGetTokens } from "./utils";
+import * as cookieParser from "cookie-parser";
+
+//Logged in user is Jessica Williamson for admin routes /teams and /teams/voyages/:voyageid
+//Logged in user is Dan ko for team member routes /teams/:teamid and /teams/:teamid/members
+//Dan Ko is part of the team with team id 4
+
+describe("Teams Controller (e2e)", () => {
+    let app: INestApplication;
+    let prisma: PrismaService;
+    let accessToken: any;
+
+    beforeAll(async () => {
+        const moduleFixture: TestingModule = await Test.createTestingModule({
+            imports: [AppModule],
+        }).compile();
+
+        await seed();
+        app = moduleFixture.createNestApplication();
+        prisma = moduleFixture.get<PrismaService>(PrismaService);
+        app.useGlobalPipes(new ValidationPipe());
+        app.use(cookieParser());
```

Comment:
Should we add an `await` before `moduleFixture.createNestApplication()`,`app.useGlobalPipes(new ValidationPipe())`, and `app.use(cookieParser())` to ensure that they complete before the application is initialized?

Reply:
are they async functions? I've actually never seen them used with await even in the nestjs docs

Reply:
We don't need to add await  as they are not promises , they are all synchronous operations and `app.use(cookieParser())` is a middleware for parsing cookies in our http request for all endpoints.